CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -0,0 +1,259 @@
+			game.Players.LocalPlayer.Character.Humanoid:EquipTool(tool)
+		end
+	end)
+	slot.OnEnter = function()
+		slot.icon.BorderSizePixel = 4
+	end
+	slot.OnLeave = function()
+		slot.icon.BorderSizePixel = 0
+	end
+--	slot.icon.MouseEnter:connect(slot.OnEnter)
+--	slot.icon.MouseLeave:connect(slot.OnLeave)
+	
+	DoLayout()
+end
+
+local function OnChildAdded(child)
+	if child:IsA("Tool") or child:IsA("HopperBin") then
+		if not Tools[child] then
+			AddTool(child)
+		end
+	end
+end
+
+local function RemoveTool(tool)
+	if not (Tools[tool] and Tools[tool].icon) then
+		return
+	end
+	Tools[tool].icon:Destroy()
+	for i, v in ipairs(ToolsList) do
+		if v == tool then
+			table.remove(ToolsList, i)
+			break
+		end
+	end
+	Tools[tool] = nil
+	DoLayout()
+end
+
+local function OnChildRemoved(child)
+	if child:IsA("Tool") or child:IsA("HopperBin") then
+		if Tools[child] then
+			print(child, child.Parent)
+			if child.Parent ~= player.Backpack and child.Parent ~= player.Character then
+				print("remove")
+				RemoveTool(child)
+			end
+		end
+	end
+end
+
+local function OnCharacterAdded(character)
+	player:WaitForChild("Backpack")
+	for i, v in ipairs(ToolsList) do
+		RemoveTool(v)
+	end
+	Tools = {}
+	ToolsList = {}
+	
+	character.ChildAdded:connect(OnChildAdded)
+	character.ChildRemoved:connect(OnChildRemoved)
+	hopperbinGUI.Parent = player.PlayerGui
+	
+	for i, v in ipairs(player:WaitForChild("Backpack"):GetChildren()) do
+		OnChildAdded(v)
+	end
+	
+	player.Backpack.ChildAdded:connect(OnChildAdded)
+	player.Backpack.ChildRemoved:connect(OnChildRemoved)
+end
+
+player.CharacterAdded:connect(OnCharacterAdded)
+if player.Character then
+	OnCharacterAdded(player.Character)
+end
+
+local zeroVector = Vector3.new(0, 0, 0)
+local horizontalRotation = CFrame.new()
+game:GetService("RunService"):BindToRenderStep("HopperBin3D", Enum.RenderPriority.Last.Value, function()
+	local cameraCFrame = workspace.CurrentCamera:GetRenderCFrame()
+	local cameraLook = cameraCFrame.lookVector
+	local cameraHorizontalVector = Vector3.new(cameraLook.X, 0, cameraLook.Z).unit
+
+	local cameraPitchAngle = math.asin(cameraLook.Y)
+	if cameraPitchAngle > math.rad(-25) then
+		local cameraHorizontalRotation = CFrame.new(zeroVector, cameraHorizontalVector)
+		horizontalRotation = cameraHorizontalRotation
+	end
+	
+	local position = workspace.CurrentCamera.CFrame.p	
+
+	local verticalError = math.abs((cameraPitchAngle - HOPPERBIN_ANGLE) / verticalRange)
+	local horizontalError = math.acos(cameraHorizontalVector:Dot(horizontalRotation.lookVector)) / horizontalRange
+	
+	SetTransparency(math.max(verticalError, horizontalError))
+	
+	local hopperbinVector = HOPPERBIN_ROTATION:vectorToWorldSpace(HOPPERBIN_OFFSET)
+	hopperbinVector = horizontalRotation:vectorToWorldSpace(hopperbinVector)
+	hopperbinPart.CFrame = CFrame.new(position + hopperbinVector, position)
+end)
+
+local image = Instance.new("ImageLabel", hopperbinGUI)
+image.Image = "http://www.roblox.com/asset/?id=286708119"
```

Comment:
Was a random placeholder. Can we get a designer on it or should we just pull the texture I had used?